core/src/main/java/org/eclipse/hono/util/ResourceLimits.java

```diff
@@ -55,13 +57,35 @@ public final ResourceLimits setMaxConnections(final int maxConnections) {
     /**
      * Gets the maximum number of connected devices a tenant supports.
      * 
-     * @return The maximum number of connections or {@link TenantConstants#DEFAULT_MAX_CONNECTIONS}
+     * @return The maximum number of connections or {@link TenantConstants#UNLIMITED_CONNECTIONS}
      *         if not set.
      */
     public final int getMaxConnections() {
         return this.maxConnections;
     }
 
+    /**
+     * Sets the maximum time-to-live to use for events published by
+     * devices of a tenant.
+     * 
+     * @param maxTtl The time-to-live in seconds.
+     * @return A reference to this for fluent use.
+     */
+    public ResourceLimits setMaxTtl(final long maxTtl) {
```

Comment:
I was wondering on how to deal with a maxTtl value of 0.
The Proton MessageImpl.setTtl(0) implementation seems inconsistent here (such a header value only being set if there is already a header set).
Of course the main question is whether there is a use case for this. It could potentially be used to effectively disable the event transport (temporarily), while still allowing telemetry - provided Artemis really uses the 0 value there to discard message directly.
@sophokles73 Do you think it would be worth it to consider such a case, testing it and making the behaviour for maxTtl=0 consistent?

Reply:
my understanding is that proton won't set a message's ttl to 0, but instead use this value to *clear* an existing value, i.e. remove the ttl property altogether. A TTL doesn't make much sense because it would mean that the receiver would discard the message immediately on reception, right?
So, no, I don't think that we need to spend much effort on it ...

Reply:
I didn't see a 0 ttl value excluded from being put into the amqp message headers on the wire, neither it being ignored in Artemis. And my guess towards a use case was explicitly based upon Artemis potentially discarding all such messages directly.
But, that's just a guess from the code and I haven't tested it.
And yes, it's probably not worth the effort.